pkg/scheduler/framework/v1alpha1/types.go

```diff
@@ -50,16 +51,23 @@ type PodInfo struct {
 	InitialAttemptTimestamp time.Time
 }
 
-// DeepCopy returns a deep copy of the PodInfo object.
-func (podInfo *PodInfo) DeepCopy() *PodInfo {
-	return &PodInfo{
-		Pod:                     podInfo.Pod.DeepCopy(),
-		Timestamp:               podInfo.Timestamp,
-		Attempts:                podInfo.Attempts,
-		InitialAttemptTimestamp: podInfo.InitialAttemptTimestamp,
+// DeepCopy returns a deep copy of the PodQueueInfo object.
+func (pqi *PodQueueInfo) DeepCopy() *PodQueueInfo {
+	return &PodQueueInfo{
+		Pod:                     pqi.Pod.DeepCopy(),
+		Timestamp:               pqi.Timestamp,
+		Attempts:                pqi.Attempts,
+		InitialAttemptTimestamp: pqi.InitialAttemptTimestamp,
 	}
 }
 
+// PodInfo is a wrapper to a Pod with additional pre-computed information to
+// accelerate processing. This information is typically immutable (e.g., pre-processed
+// inter-pod affinity selectors).
+type PodInfo struct {
```

Comment:
just to confirm... we are not using this type yet, right?

Reply:
We are, it is the type used by NodeInfo.Pods/NodeInfo.PodsWithAffinity. I will follow up with two PRs: 1) add the first pre-computed data (the pre-processed selectors), 2) change PodLister interface to PodInfoLister.